ibis/backends/base/sql/__init__.py

```diff
@@ -344,6 +345,7 @@ def compile(
             The output of compilation. The type of this value depends on the
             backend.
         """
+        util.consume(self._compile_udfs(expr))
```

Comment:
Why are we compiling the UDFs here (and ignoring the results)? Just so we error here if the UDF can't be compiled?

Reply:
What if you want to call `.alias` on an expression that uses a UDF? How would an engine be able to infer the types of the query without knowing about UDFs?

Reply:
Ah right, this should be unnecessary by registering the generic `ScalarUDF` dispatcher as you describe below. That's the reason why I was doing it here!

Reply:
This will have to stay in light of rule lookup not taking into account inheritanc.e